src/backend/mod.rs

```diff
@@ -120,6 +124,27 @@ impl DNSBackend {
         Some(results)
     }
 
+    // Checks if a container is associated with only internal networks.
+    // Returns true iff a container is only present in internal networks.
+    pub fn ctr_is_internal(&self, requester: &IpAddr) -> bool {
+        let mut is_internal = true;
+
+        match self.ip_mappings.get(requester) {
+            Some(nets) => {
+                for net in nets {
+                    match self.network_is_internal.get(net) {
+                        Some(b) => is_internal &= b,
+                        None => continue,
+                    }
+                }
+            }
+            // If we don't know about you, you don't get DNS
+            None => return true,
```

Comment:
Isn't this a breaking change? I think it makes sense but AFAIK we currently allow requests from the host to do that so that can break people I think.
While I agree it would be best to have this I am not sure to include if we decide to backport

Reply:
Hm. False is probably safer in that case. I don't have a strong opinion here and the risk of breaking means we should avoid this.